cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
+	end = regexp.MustCompile(`(?i)\s+(-?(?:filetype|package|pkg|path|file)):(\S+)\s*$`)
+)
+
+func parseFilters(query *url.Values, qstr string, atstart bool) string {
```

Comment:
I think a better function signature would be `func rewriteFilters(query url.Values, filtersRe *regexp.Regexp) url.Values`

Then, you can call it using `query = rewriteFilters(query, start)` and `query = rewriteFilters(query, end)`